sources/packages/backend/apps/api/src/route-controllers/restriction/_tests_/e2e/restriction.institution.controller.getStudentRestrictionDetail.e2e-spec.ts

```diff
@@ -0,0 +1,154 @@
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+    applicationRepo = appDataSource.getRepository(Application);
+  });
+
+  it("Should return forbidden 403 error code when a non BC Public Institution User tries to get student restriction by student restriction id.", async () => {
+    // Arrange
+    const { institution } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    collegeC = institution;
+    collegeCLocation = createFakeInstitutionLocation(collegeC);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+    const student = await saveFakeStudent(appDataSource);
+    const application = createFakeApplication({
+      location: collegeCLocation,
+      student,
+    });
+    await applicationRepo.save(application);
+    const studentRestriction = await saveFakeStudentRestriction(
+      appDataSource,
+      { student, application, creator: student.user },
+      {
+        restrictionCategory: "Verification",
+        notificationType: RestrictionNotificationType.NoEffect,
+      },
+    );
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    const endpoint = `/institutions/restriction/student/${student.id}/studentRestriction/${studentRestriction.id}`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.FORBIDDEN);
+  });
+
+  it("Should return forbidden 403 error code when a BC Public Institution User tries to get student restriction for a student which they don't have access to.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(appDataSource);
+    const studentRestriction = await saveFakeStudentRestriction(
+      appDataSource,
+      { student, creator: student.user },
+      {
+        restrictionCategory: "Verification",
+        notificationType: RestrictionNotificationType.NoEffect,
+      },
+    );
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const endpoint = `/institutions/restriction/student/${student.id}/studentRestriction/${studentRestriction.id}`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.FORBIDDEN);
+  });
+
+  it("Should get the student restriction when a BC Public Institution User gets a student restriction for a student they have access to.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(appDataSource);
+    const application = createFakeApplication({
+      location: collegeFLocation,
+      student,
+    });
+    await applicationRepo.save(application);
+    const studentRestriction = await saveFakeStudentRestriction(appDataSource, {
```

Comment:
As per my understanding the method `restrictionRepo.findOne` from `saveFakeStudentRestriction` is not controlling the restriction that will be associated with the student. How can we ensure that it will not be a "No effect" which would lead to no records returned?

Reply:
When I getting the student restriction by Student Restriction Id, I am not filtering it. It can return any restriction because the controller / service method `getStudentRestrictionDetail` for this test scenario does not do filtering of restrictions.

Reply:
@sh16011993 if the institution cannot list the restriction, the institution cannot get it. The method `getStudentRestrictionDetail` should not allow the institution to retrieve a restriction type if it does not have access to it.